features/upgrade/monitoring/monitoring-upgrade.feature

```diff
@@ -0,0 +1,82 @@
+Feature: cluster monitoring related upgrade check
+  # @author hongyli@redhat.com
+  @upgrade-prepare
+  @admin
+  Scenario: upgrade cluster monitoring along with OCP - prepare
+    Given I switch to cluster admin pseudo user
+    When I run the :apply client command with:
+      | f          | <%= BushSlicer::HOME %>/features/upgrade/monitoring/testdata/cm-monitoring-retention.yaml |
+      | overwrite  | true |
+    Then the step should succeed
+
+  # @author hongyli@redhat.com
+  # @case_id OCP-29797
+  @upgrade-check
+  @admin
+  Scenario: upgrade cluster monitoring along with OCP
+    Given I switch to cluster admin pseudo user
+    And I use the "openshift-monitoring" project
+    When I run the :get client command with:
```

Comment:
@lihongyan1 you can replace line 19-25 with `Given all pods in the project are ready`